addons/im_livechat_mail_bot/controllers/discuss.py

```diff
@@ -0,0 +1,13 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo.http import request, route
+from odoo.addons.mail.controllers import discuss
+
+
+class DiscussController(discuss.DiscussController):
+    @route()
+    def mail_message_post(self, thread_model, thread_id, post_data, **kwargs):
+        if post_data.get("canned_response_ids"):
+            request.env.context = dict(request.env.context, canned_response_ids=post_data["canned_response_ids"])
```

Comment:
Shouldn't this use `request.update_context` instead?

Reply:
It's not available in V15.0